specifyweb/specify/sp7_build_models.py

```diff
@@ -0,0 +1,275 @@
+import logging
+from . import model_extras
+
+appname = __name__.split('.')[-2]
+logger = logging.getLogger(__name__)
+
+TAB = '    '
+NEXT = ', '
+CLOSE = ')\n'
+
+SPECIAL_DELETION_RULES_CODE = {
+    'Agent.specifyuser': 'models.SET_NULL',
+    'Recordsetitem.recordset': 'models.CASCADE',
+
+    # Handle workbench deletion using raw sql in business rules.
+    'Workbenchrow.workbench': 'models.DO_NOTHING',
+    'Workbenchdataitem.workbenchrow': 'models.DO_NOTHING',
+    'Workbenchrowimage.workbenchrow': 'models.DO_NOTHING',
+    'Workbenchrowexportedrelationship.workbenchrow': 'models.DO_NOTHING',
+
+    'Spappresourcedir.specifyuser': 'models.CASCADE',
+    'Spappresource.specifyuser': 'models.CASCADE',
+    'Spappresource.spappresourcedir': 'models.CASCADE',
+    'Spappresourcedata.spappresource': 'models.CASCADE',
+    'Spappresourcedata.spviewsetobj': 'models.CASCADE',
+    'Spreport.appresource': 'models.CASCADE',
+}
+
+FIELD_TYPE_CODE_MAP = {
+    'java.lang.String': 'models.CharField',
+    'java.util.Calendar': 'models.DateTimeField',
+    'java.util.Date': 'models.DateTimeField', # This mapping wasn't in the original code
+    'java.sql.Timestamp': 'models.DateTimeField',
+    'java.lang.Integer': 'models.IntegerField',
+    'java.lang.Long': 'models.BigIntegerField',
+    'java.lang.Byte': 'models.SmallIntegerField',
+    'java.lang.Short': 'models.SmallIntegerField',
+    'java.math.BigDecimal': 'models.DecimalField',
+    'java.lang.Double': 'models.FloatField', # This mapping wasn't in the original code
+    'java.lang.Boolean': 'models.BooleanField',
+    'text': 'models.TextField',
```

Comment:
Feel free to shot it down. there is a discrepancy in how datamodel defines spappresourcedata. It is actually a BLOB but treated as a TextField in django, which required ugly workaround in record merging. Could you look at schema (in mariadb) and figure out more cases like that, and possibly resolve them?

Reply:
Good idea, I'll look into that

Reply:
I started work on the blob fields in this branch https://github.com/specify/specify7/tree/issue-4358-blobs, still a work in progress, most of the workbench upload unit tests failed.

Reply:
ok, in this branch https://github.com/specify/specify7/tree/issue-4358-blobs, I changed the django fields from `TextField` to `BinaryField` where the mysql field is `mediumblob`.  I then added functions to override the save function for those fields to make sure to encode the data before saving it to the database.  I also added a `get_decoded_data` function as well.

@realVinayak do these changes suffice for avoiding issues in record merging?  Is there a unit test for this edge case?